localstack/extensions/patterns/webapp.py

```diff
@@ -0,0 +1,321 @@
+import importlib
+import logging
+import mimetypes
+import typing as t
+from functools import cached_property
+
+from rolo.gateway import HandlerChain
+from rolo.router import RuleAdapter, WithHost
+from werkzeug.routing import Submount
+
+from localstack import config
+from localstack.aws.api import RequestContext
+from localstack.extensions.api import Extension, http
+
+if t.TYPE_CHECKING:
+    # make sure jinja2 stays optional
+    import jinja2
```

Comment:
might be worth commenting that `jinja2` is a peer dependency that needs to be installed through the extension for templating